pkg/gitproviders/gitprovidersfakes/fake_git_provider_handler.go

```diff
@@ -460,6 +486,124 @@ func (fake *FakeGitProviderHandler) GetAccountTypeReturnsOnCall(i int, result1 g
 	}{result1, result2}
 }
 
+func (fake *FakeGitProviderHandler) GetUser(arg1 context.Context, arg2 *oauth2.Token) (*gitproviders.User, error) {
+	fake.getUserMutex.Lock()
+	ret, specificReturn := fake.getUserReturnsOnCall[len(fake.getUserArgsForCall)]
+	fake.getUserArgsForCall = append(fake.getUserArgsForCall, struct {
+		arg1 context.Context
+		arg2 *oauth2.Token
+	}{arg1, arg2})
+	stub := fake.GetUserStub
+	fakeReturns := fake.getUserReturns
+	fake.recordInvocation("GetUser", []interface{}{arg1, arg2})
+	fake.getUserMutex.Unlock()
```

Comment:
Can you elaborate on unlocking here instead of a defer Unlock() right after taking the lock?

Reply:
This is generated code.